src-tauri/src/main.rs

```diff
@@ -107,6 +120,7 @@ async fn connect_to_serial_port(
     let handle = app_handle.app_handle().clone();
     let mesh_device_arc = mesh_device.inner.clone();
     let mesh_graph_arc = mesh_graph.inner.clone();
+    let algo_state_arc = algo_state.inner.clone();
```

Comment:
Something that just came to mind (cleanliness, feel free to bump to a new issue or ignore), it might be worth adding a new command that initializes the graph and the algos. The `connect_to_serial` command ideally wouldn't initialize anything that it doesn't need to function, which I would say includes the algos and graph

Reply:
Resolving conversation by defaulting to `save` to `false` and initializing data structures in main as `None`.

Reply:
Added a new function, `initialize_graph_state`, and moved arc/mutex data structure initialization there.